### PR TITLE
Question Sharing documentation for Configuring Sharing in JSON

### PR DESCRIPTION
**Pull request by SethPoulsen**
Update documentation about how to update sharing info in JSON files, in preparation for getting question sharing into public access.

### CODE REVIEW

**Comment by SethPoulsen** [comment]
Maybe we want to merge the changes talk about JSON configuration, then remove the message for exiting Beta separately? 

Or we can do both changes at once, whatever you think is best @nwalters512

**Comment by SethPoulsen** [comment]
I have marked as 'do not merge' for now because it shouldn't be merged until #10029 is also merged.

File: docs/course.md
Comment (nwalters512) [inline docs/course.md:389]:
```suggestion

```

File: docs/course.md
Comment (nwalters512) [inline docs/course.md:390]: Nit:

```suggestion
At this time, sharing sets cannot be renamed or removed from `infoCourse.json` once they have been added, because deleting a sharing set shared to another course could break assessments in that course.
```

File: docs/question.md
Comment (nwalters512) [inline docs/question.md:56]: IMO don't include this here.

```suggestion
```

File: docs/course.md
Comment (nwalters512) [inline docs/course.md:364]:
```suggestion
Questions can be added to sharing sets to enable other courses to use your questions in their assessments. Sharing sets are designed for sharing questions only to specific courses, if you would like to make questions available for anyone to uses, you should make them publicly shared as explained in the [question documentation](question.md), or the [sharing documentation](questionSharing.md).
```

File: docs/question.md
Comment (nwalters512) [inline docs/question.md:100]:
```suggestion
Any question that is marked with `"sharePublicly": true` or `"shareSourcePublicly": true` will be considered and displayed as being published for free use under the [CC-BY-NC](https://www.creativecommons.org/licenses/by-nc/4.0/) license. Questions may be privately shared to individual courses using sharing sets, as explained in the [sharing documentation](questionSharing.md). Sharing sets that a question belongs to are specified as a list of strings. These must match sharing sets that are declared in the [course configuration](course.md#adding-sharing-sets).
```

File: docs/questionSharing.md
Comment (nwalters512) [inline docs/questionSharing.md:11]:
```suggestion
Questions can either be shared publicly, so that anyone can preview the questions and use them in their course, or shared only to specific other courses using sharing sets.
```

File: docs/questionSharing.md
Comment (nwalters512) [inline docs/questionSharing.md:13]: Nit:

```suggestion
Any question that is marked with `"sharePublicly": true` or `"shareSourcePublicly": true` will be considered and displayed as being published for free use under the [CC-BY-NC license](https://www.creativecommons.org/licenses/by-nc/4.0/).
```

File: docs/questionSharing.md
Comment (nwalters512) [inline docs/questionSharing.md:17]:
```suggestion
Access to shared questions which are not shared publicly is controlled through **sharing sets**. A sharing set is a named set of questions which you can share to another course all at once. The sharing set system exists so that course owners may differentially share different sets of their questions. For example, an instructor may want to share some questions only with other courses in their department, and other questions with anyone using PrairieLearn. Sharing sets are created by adding them to the [`infoCourse.json` file](course.md#sharing-sets).
```

File: docs/questionSharing.md
Comment (nwalters512) [inline docs/questionSharing.md:71]: Nit: let's be consistent with either ending list items with periods, or not doing so. I'd be inclined to add periods everywhere.

**Review by nwalters512 — APPROVED** [review]
I won't hard-block on any of these.

File: docs/questionSharing.md
Comment (SethPoulsen) [reply docs/questionSharing.md:71]: done

File: docs/questionSharing.md
Comment (nwalters512) [inline docs/questionSharing.md:70]: Extremely a nit, but on pretty much every other page but this one we use double quotes for things like this. Grep for `[a-zA-Z] '[a-zA-Z]` and `[a-zA-Z] "[a-zA-Z]` in `docs/` and compare.

**Pull request merged**
